### ISSUE TITLE
Removing `.map` parens breaks build with `?`

### ISSUE THREAD

**Issue by keith** [nicklockwood/SwiftFormat#547]
With version 0.44.1

When you have this code:

```swift
func foo(_ bar: Int) -> Int? {
    return bar
}

let a: Int? = 1
let b = a.map({ foo($0) })?.map({ foo($0) })
print("\(String(describing: b))")
```

And run `swiftformat foo.swift`, you end up with:

```swift
func foo(_ bar: Int) -> Int? {
    return bar
}

let a: Int? = 1
let b = a.map { foo($0) } ?.map { foo($0) }
print("\(String(describing: b))")
```

Which is no longer valid because of the space before `?`. Removing the space fixes it.

**Comment by nicklockwood** [comment]
@keith thanks, I'll fix it.

**Comment by nicklockwood** [comment]
@keith fixed in 0.44.2.
